src/gui/editorwidgets/qgsdefaultsearchwidgetwrapper.h

```diff
@@ -41,7 +41,7 @@ class GUI_EXPORT QgsDefaultSearchWidgetWrapper : public QgsSearchWidgetWrapper
     void setExpression( QString exp ) override;
 
   private slots:
-    void setCaseString( Qt::CheckState );
+    void setCaseString( int caseSensitiveCheckState );
```

Comment:
I noticed an error on my system (and I also think it occurs in different platform) this slot never was called because the signature object is different from that of the signal emitted. You should get a warning like this:
```Object::connect: No such slot QgsDefaultSearchWidgetWrapper::setCaseString( int )``` and casesensitive checkbox does not work as expected.